skimage/segmentation/chanvese.py

```diff
@@ -0,0 +1,175 @@
+import numpy as np
+from scipy.ndimage import distance_transform_cdt
+
+
+def _cv_curvature(phi):
+    """ Returns the 'curvature' of a level set 'phi'.
+    """
+    m, n = phi.shape
+    P = np.hstack((np.zeros((m, 1)), phi, np.zeros((m, 1))))
+    m, n = P.shape
+    P = np.vstack((np.zeros((1, n)), P, np.zeros((1, n))))
+    fy = P[2:, 1:-1] - P[0:-2, 1:-1]
+    fx = P[1:-1, 2:] - P[1:-1, 0:-2]
+    fyy = P[2:, 1:-1] + P[0:-2, 1:-1] - 2 * phi
+    fxx = P[1:-1, 2:] + P[1:-1, 0:-2] - 2 * phi
```

Comment:
Lines 12-15 probably could be refactored using `np.diff()`. OK tor me as it is, though.